pcl_utilities/src/tests/simple_test_concatenate_point_cloud.cpp

```diff
@@ -0,0 +1,181 @@
+    integer_range.to_value = MAX_POINT_CLOUDS;
+    param_constraints.integer_range.push_back(integer_range);
+
+    num_point_clouds_ = static_cast<size_t>(
+      declare_parameter<int64_t>("num_point_clouds", std::move(param_constraints)));
+
+    concatenate_point_cloud_client_ =
+      create_client<PCLConcatenatePointCloud>(client_topic);
+
+    output_publisher_ = create_publisher<PointCloud2>(
+      "concatenate_point_cloud/cloud_concatenated", 1);
+
+    history_ = std::deque<PointCloud2>();
+  }
+
+  void spin()
+  {
+    while(rclcpp::ok())
+    {
+      PointCloud2 point_cloud_message;
+      bool was_retrieved = rclcpp::wait_for_message(
+        point_cloud_message, shared_from_this(),
+        camera_topic_, MAX_WAIT_TIME);
+
+      if (!was_retrieved)
+      {
+        RCLCPP_ERROR(get_logger(),
+          "A camera message could not be retrieved within 1 second.");
+        continue;
+      }
+
+      process_point_cloud(point_cloud_message);
+    }
+  }
+
+  void process_point_cloud(PointCloud2& point_cloud) {
+    // create a fixed sized context window of old point clouds
+    history_.push_back(std::move(point_cloud));
+    if (history_.size() > num_point_clouds_)
+    {
+      history_.pop_front();
+    } else if (history_.size() < num_point_clouds_)
+    {
+      return;
+    }
+
+    auto request = std::make_shared<PCLConcatenatePointCloud::Request>();
+    request->cloud_list_in = std::vector(history_.cbegin(), history_.cend());
+
+    auto response_future =
+      concatenate_point_cloud_client_->async_send_request(request);
+
+    auto response_code = rclcpp::spin_until_future_complete(
+      shared_from_this(), response_future, MAX_WAIT_TIME);
+
+    if (response_code != rclcpp::FutureReturnCode::SUCCESS)
+    {
+      RCLCPP_ERROR(get_logger(), "Failed to recieve a response from the service");
+      return;
+    }
+
+    PCLConcatenatePointCloud::Response::SharedPtr response {
+      response_future.get()};
+    output_publisher_->publish(response->cloud_out);
+
+    test_point_cloud(request, response);
+  }
+
+  void test_point_cloud(
+    PCLConcatenatePointCloud::Request::SharedPtr,
+    PCLConcatenatePointCloud::Response::SharedPtr response)
+  {
+    std::stringstream output_stream;
+    output_stream << std::fixed << std::setprecision(2);
+
+    // PCL takes the maximum timestamp when concatenating messages
+    // The time stamp of the newest message passed to the service.
+    double current_time = rclcpp::Time(response->cloud_out.header.stamp).seconds();
+
+    // The folling tests are stand-ins for future unit tests
+    // ASSERTIONS:
+    //  - The ouput point cloud must be at least the size of the
+    //    largest input cloud.
+    std::size_t max_point_cloud_size = 0U;
+    for (const auto& point_cloud : history_) {
+        max_point_cloud_size = std::max(max_point_cloud_size, point_cloud.data.size());
+    }
+
+    if (
+      response->cloud_out.data.size() < max_point_cloud_size)
+    {
```

Comment:
This should not be split on two lines.

Once it is on one line, brace should cuddle on if: https://docs.ros.org/en/humble/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html#open-versus-cuddled-braces